kernel/src/lib.rs

```diff
@@ -62,14 +58,14 @@ pub fn main<P: Platform, C: Chip>(
 
             for (i, p) in processes.iter_mut().enumerate() {
                 p.as_mut().map(|process| {
-                    sched::do_process(platform, chip, process, AppId::new(i), ipc);
+                    sched::do_process(platform, chip, process, callback::AppId::new(i), ipc);
                 });
                 if chip.has_pending_interrupts() {
                     break;
                 }
             }
 
-            support::atomic(|| {
+            chip.atomic(|| {
```

Comment:
Feels like the implementation of atomic may be chip-specific, but the notion of an atomic operation should be generic to the kernel?

Reply:
`atomic` is included in the `Chip` trait, so the kernel does enforce that atomic has to exist for the chip, and that way we use the correct atomic version for the chip.